packages/core/src/components/progress/progress.ts

```diff
@@ -0,0 +1,7 @@
+export interface ProgressProps {
+  children?: string;
```

Comment:
No need to set `children`, it's a part of element props.

```suggestion
```

Reply:
I have to set it as string though, if not, typescript complains for the `aria-valuetext` prop.

[About the `aria-valuetext`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_progressbar_role#example_2_using_aria-valuetext)

There are other options I can think of:
 - Remove the `aria-valuetext` and let this up to the end user to add
 - Use a string prop `label` instead of `children`

What do you think is best?

Reply:
I like `children`, although it's a React convention I don't mind other (future) rendering engines calling it children either.
We can always change the interface specific to a single engine if needed.

I'm fine with setting it to `string` here.

Reply:
Actually, I take that back, that's no good, we lose the ability to compose, say:

```tsx
<Progress value={22} value={40}>
  Step <MyCode>22</MyCode> / <MyCode>40</MyCode>
</Progress>
```

Let's not define it as `string`, rather we will need more runtime code in React's component, e.g.

```tsx
  aria-valuetext={typeof children === 'string' ? children : undefined}
```

It'd be ideal if we could read the `textContent` of `children` that are `JSX.Element`s but that might require too much code, with no real benefit.

Reply:
I have changed it to:

```
aria-valuetext={
        ariaValuetext || (typeof children === 'string' ? children : undefined)
}
```
So the user can also provide the `aria-valuetext` himself, what do you think?

Reply:
Yeap that looks good.

Lazy question because I don't feel like reading the entire w3 document again, can't it infer the `valuetext` from the `textContent` of the element?

Reply:
I think we have to fill it when we change the default (which is a percentage number).

> Authors SHOULD only set the `aria-valuetext` attribute when the rendered value cannot be meaningfully represented as a number.

https://www.w3.org/TR/wai-aria-1.1/#aria-valuetext

Reply:
Cool so it'd read `valuenow` incorrectly if the `valuemax` is different. Let's leave it then.